src/move/econia/sources/market.move

```diff
@@ -5007,9 +5168,28 @@ module econia::market {
             price, restriction, self_match_behavior, &custodian_capability);
         // Check if order is tail of corresponding price level.
         assert!(is_local_tail_test(market_id, side, market_order_id), 0);
+        // Assert event streams.
+        assert!(user::get_cancel_order_events_test(
```

Comment:
Why assert the cancel events here if we're only changing the order size? Seems kind of random. Do orders self-cancel as a result of a size change to a size too low? Is that tested, and either way I feel like trying to set an order to a size too low should just throw exception instead of canceling.

Docs don't mention any cancel-by-size-change: https://econia.dev/off-chain/events#changing-order-size

Reply:
Cancel order events are asserted as empty here because user-specific size changes are handled internal to user memory as a cancel followed by an immediate placement

This is the rationale for having `user::CANCEL_REASON_SIZE_CHANGE_INTERNAL`, and is related to #360

Asserting there are no cancel events is done to enforce that a cancel event isn't erroneously emitted due to the user-side logic